app/__init__.py

```diff
@@ -106,6 +107,14 @@ def create_app(application, config=None):
     request_helper.init_app(application)
     db.init_app(application)
     migrate.init_app(application, db=db)
+
+    # Patch for flask-marshmallow compatibility with Flask-SQLAlchemy 2.3.2.dev
+    if "sqlalchemy" in application.extensions:
```

Comment:
We had to update the `flask-marshmallow` and `Marshmallow` dependencies to support `flask-smorest`. The newer version of `flask-marshmallow` expects the Flask app's extension registry to have a `session` attribute on `app.extensions["sqlalchemy"]`. 

Since we've [pinned `Flask-SQLAlchemy'](https://github.com/cds-snc/notification-api/blob/cbcd5018222fb06912967937030b673b34e0f37e/pyproject.toml#L42-L43) to a specific version there's a disagreement between the libraries that we need to patch in order for the extension to be registered correctly.